installers/custom_ipxe/main.go

```diff
@@ -3,22 +3,71 @@ package custom_ipxe
 import (
 	"strings"
 
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/boots/installers"
 	"github.com/tinkerbell/boots/ipxe"
 	"github.com/tinkerbell/boots/job"
+	"github.com/tinkerbell/boots/packet"
 )
 
 func init() {
-	job.RegisterSlug("custom_ipxe", bootScript)
+	job.RegisterInstaller("custom_ipxe", ipxeScript)
+	job.RegisterSlug("custom_ipxe", ipxeScript)
 }
 
-func bootScript(j job.Job, s *ipxe.Script) {
+func ipxeScript(j job.Job, s *ipxe.Script) {
+	logger := installers.Logger("custom_ipxe")
+	if j.InstanceID() != "" {
```

Comment:
Hmm we should probably return a s.Shell if there's no instance id right?

Reply:
I see your point, will update.